hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/OzoneManagerProtocolClientSideTranslatorPB.java

```diff
@@ -292,13 +295,30 @@ private RetryAction getRetryAction(RetryAction fallbackAction,
     return proxy;
   }
 
+  /**
+   * Unwrap exception to check if it is a {@link AccessControlException}.
+   */
+  private boolean isAccessControlException(Exception ex) {
+    if (ex instanceof ServiceException) {
+      Throwable t = ex.getCause();
+      if (t instanceof IOException) {
+        t = t.getCause();
+        if (t instanceof IOException) {
```

Comment:
Wow is it wrapped that deeply? Can we make this check a bit more robust, if any of the nested exceptions is an instanceOf `AccessControlException` and also check in a loop while `getCause` returns non-nill?

Reply:
null check is covered by instanceof operator which returns false if ex is null.
Making a change for checking ACE on each level.

Reply:
I meant - can we keep checking for ACE until we finally get null? I am not sure if any server side changes could impact how the exception is wrapped e.g. introduce more nesting.

Reply:
Oh yes, doing the same, thanks!